pkg/scheduler/framework/plugins/defaultpreemption/default_preemption_test.go

```diff
@@ -231,8 +231,10 @@ func TestPostFilter(t *testing.T) {
 			}
 
 			gotResult, gotStatus := p.PostFilter(context.TODO(), state, tt.pod, tt.filteredNodesStatuses)
-			if !reflect.DeepEqual(gotStatus, tt.wantStatus) {
-				t.Errorf("Status does not match: %v, want: %v", gotStatus, tt.wantStatus)
+			if diff := cmp.Diff(gotStatus, tt.wantStatus,cmp.Comparer(func(x,y framework.Status) bool{
+				return reflect.DeepEqual(x,y)
```

Comment:
I suppose the goal of this PR is to eliminate `refelect.DeepEqual()`, so why introduce it here again?

Reply:
try:

```go
if diff := cmp.Diff(gotStatus, tt.wantStatus, cmp.AllowUnexported(framework.Status{})); diff != "" {
```

Reply:
I would avoid using `AllowUnexported`. We have two options:

1. Implement `Status.Compare(*Status)`
2. Export all fields and use https://godoc.org/github.com/google/go-cmp/cmp/cmpopts#EquateErrors

I prefer 2, but open to debate

Reply:
I actually said this in the original issue #94696

Reply:
Did you mean use `EquateErrors` like this?
```
if diff:= cmp.Diff(gotStatus,tt.wantStatus,cmpopts.EquateErrors(),cmp.Exporter(func(r reflect.Type) bool {
	return true
})); diff != "" {
```